org.adempiere.base/src/org/compiere/model/MZoomCondition.java

```diff
@@ -244,16 +252,33 @@ public static int findZoomWindowByWindowId(int AD_Window_ID, MQuery query, int w
 						window.initTab(parentTab.getTabNo());					
 						if (parentTab.getParentTab() != null)
 						{
-							parentId = DB.getSQLValue(null, "SELECT " + parentTab.getLinkColumnName() + " FROM " + parentTab.getTableName() + " WHERE " 
-									+ parentTab.getTableName()+"_ID="+parentId);
-							if (parentId <= 0) return 0;
+							StringBuilder sql = new StringBuilder("SELECT ").append(parentTab.getLinkColumnName())
+									.append(" FROM ").append(parentTab.getTableName())
+									.append(" WHERE ");
+							if (parentTab.getLinkColumnName().endsWith("_UU")) {
+								sql.append(PO.getUUIDColumnName(parentTab.getTableName())).append("=").append(DB.TO_STRING(parentId.toString()));
+							} else {
+								sql.append(parentTab.getTableName()).append("_ID=").append(parentId);
+							}
+							parentId = DB.getSQLValue(null, sql.toString());
+							if (parentTab.getLinkColumnName().endsWith("_UU")) {
+								parentId = DB.getSQLValueString(null, sql.toString());
+							} else {
+								int tmpId = DB.getSQLValue(null, sql.toString());
+								if (tmpId > 0)
+									parentId = Integer.valueOf(tmpId);
+							}
```

Comment:
duplicate call to DB.getSQLValue ?

Reply:
Thanks @hengsin - fixed all the concerns